e2e/operator_context_test.go

```diff
@@ -337,46 +364,33 @@ func (tctx *OperatorContext) createBaseResources(ctx context.Context) error {
 		if err != nil {
 			return fmt.Errorf("deserializing alertmanager manifest: %w", err)
 		}
-		switch obj := obj.(type) {
-		case *appsv1.StatefulSet:
-			obj.Namespace = tctx.namespace
-			if obj.Labels == nil {
-				obj.Labels = map[string]string{}
-			}
-			for k, v := range tctx.getSubTestLabels() {
-				obj.Labels[k] = v
-			}
-			if _, err := tctx.kubeClient.AppsV1().StatefulSets(tctx.namespace).Create(ctx, obj, metav1.CreateOptions{}); err != nil {
-				return fmt.Errorf("create alertmanager statefulset: %w", err)
-			}
-		case *corev1.Secret:
-			obj.Namespace = tctx.namespace
-			if obj.Labels == nil {
-				obj.Labels = map[string]string{}
-			}
-			for k, v := range tctx.getSubTestLabels() {
-				obj.Labels[k] = v
-			}
-			if _, err := tctx.kubeClient.CoreV1().Secrets(tctx.namespace).Create(ctx, obj, metav1.CreateOptions{}); err != nil {
-				return fmt.Errorf("create alertmanager secret: %w", err)
-			}
-		case *corev1.Service:
-			obj.Namespace = tctx.namespace
-			if obj.Labels == nil {
-				obj.Labels = map[string]string{}
-			}
-			for k, v := range tctx.getSubTestLabels() {
-				obj.Labels[k] = v
-			}
-			if _, err := tctx.kubeClient.CoreV1().Services(tctx.namespace).Create(ctx, obj, metav1.CreateOptions{}); err != nil {
-				return fmt.Errorf("create alertmanager service: %w", err)
-			}
+		obj, ok := obj.(client.Object)
+		if !ok {
+			return fmt.Errorf("unknown object: %s", string(doc))
+		}
+
+		obj.SetNamespace(tctx.namespace)
+		labels := obj.GetLabels()
+		if labels == nil {
+			labels = map[string]string{}
+		}
+		for k, v := range tctx.getSubTestLabels() {
+			labels[k] = v
+		}
+		obj.SetLabels(labels)
+
+		if err := kubeClient.Create(ctx, obj); err != nil {
+			return fmt.Errorf("create object %s: %w", string(doc), err)
```

Comment:
Is `doc` the raw yaml? That's going to end up being an oddly formatted error, I think.